tests/404_or_greater/expr_match.ml

```diff
@@ -0,0 +1,12 @@
+module M = struct
+  type t = Foo | Bar
+  type r = { i : int; t : t }
+end
+
+let f () =
+  let M.(s) = M.(Foo) in assert (s = M.Foo);
```

Comment:
I'd definitely convert all the syntax variants (`M.{}`, `M.[]`, `M.[||]`) to `match`, so that we are actually testing those, and insert an or-pattern in more of them. I'd personally delay adding `let` cases until #155... but they don't really hurt, besides making the future diff for #155 slightly less clear. But nobody is going to read that diff anyway after it's in `master`, so up to you :)